src/display/api.js

```diff
@@ -820,12 +820,22 @@ var PDFPageProxy = (function PDFPageProxyClosure() {
 
       var renderingIntent = (params.intent === 'print' ? 'print' : 'display');
       var canvasFactory = params.canvasFactory || new DOMCanvasFactory();
+
       let webGLContext = new WebGLContext({
         // TODO: When moving this parameter from `PDFJS` to {RenderParameters},
         //       change its name to `enableWebGL` instead.
         enable: !getDefaultSetting('disableWebGL'),
       });
 
+      var combinedInitialTransform;
```

Comment:
While it might be OK to have this new resizing behaviour be the default, it must be possible for an API consumer to opt-out of it (since it's a new feature).
Hence I'd suggest adding a new render parameter (with accompanying JSDoc), for example named  `resizeLargeImages`, which defaults to `true` but can be used to disable resizing.

So something like the following:
```javascript
let combinedInitialTransform = null;
if (params.resizeLargeImages !== false) {
  if (params.transform) {
    combinedInitialTransform = Util.transform(params.transform,
                                              params.viewport.transform);
  } else {
    combinedInitialTransform = params.viewport.transform;
  }
}
```